src/libcore/iter.rs

```diff
@@ -2205,6 +2297,35 @@ impl<I> Iterator for Fuse<I> where I: Iterator {
     }
 
     #[inline]
+    fn nth(&mut self, n: usize) -> Option<I::Item> {
+        if self.done {
+            None
+        } else {
+            let nth = self.iter.nth(n);
```

Comment:
Yes, at most. And that's probably reasonable, I just wanted to point out that if the underlying `nth` did not follow the spec in that sense, then this implementation and the default one could produce different results -- just want to make sure we're comfortable with that.